crates/primitives/src/receipt.rs

```diff
@@ -147,6 +147,9 @@ impl Decodable for ReceiptWithBloom {
                 } else if receipt_type == 0x02 {
                     buf.advance(1);
                     Self::decode_receipt(buf, TxType::EIP1559)
+                } else if receipt_type == 0x03 {
+                    buf.advance(1);
+                    Self::decode_receipt(buf, TxType::EIP4844)
```

Comment:
@joshieDo This just partially solve the test error. Due to the advance in the buf here, when calling the `Header::decode` function with the new type `TxType::EIP4844` we have the `DecodeError::InputTooShort` that directly comes from:

https://github.com/paradigmxyz/reth/blob/a298756d9578f32fe66e0cee2794b3a55c8f8d83/crates/rlp/src/decode.rs#L147-L149

For me this is due to the `h.payload_length` that seems too small (adapted for the old transaction types I suppose). I think this requires a change on the following line but not sure, in order to take into account the new possible `TxType::EIP4844` type:

https://github.com/paradigmxyz/reth/blob/a298756d9578f32fe66e0cee2794b3a55c8f8d83/crates/rlp/src/decode.rs#L130

Reply:
actually, what's missing is  ` | TxType::EIP4844` in 
```rust
impl<'a> Encodable for ReceiptWithBloomEncoder<'a> {
    fn encode(&self, out: &mut dyn BufMut) {
        self.encode_inner(out, true)
    }
    fn length(&self) -> usize {
        let mut payload_len = self.receipt_length();
        // account for eip-2718 type prefix and set the list
        if matches!(self.receipt.tx_type, TxType::EIP1559 | TxType::EIP2930) {
            payload_len += 1;
            // we include a string header for typed receipts, so include the length here
            payload_len += length_of_length(payload_len);
        }

        payload_len
    }
}
```

which is kinda hidden unfortunately

Reply:
added it locally, and it passed the failed test

Reply:
@joshieDo Simpler than expected, thanks, this solves the payload issue then, I've just pushed the fix.